src/components/FileUpload.js

```diff
@@ -42,6 +42,8 @@ const FileUpload = React.createClass({
   },
 
   componentWillReceiveProps (newProps) {
+    this._input.value = null;
```

Comment:
I've been getting warnings in the console when using null for input values.  We might want to use an empty string in this case instead.